models/pod.go

```diff
@@ -417,20 +418,21 @@ func IsPodTerminating(pod *v1.Pod) bool {
 	return pod.ObjectMeta.DeletionTimestamp != nil
 }
 
-// ValidatePodWaitingState returns nil if pod waiting reson is valid and error otherwise
+// ValidatePodWaitingState returns nil if pod waiting reason is valid and error otherwise
 // Errors checked:
 // - ErrImageNeverPull
 // - ErrImagePullBackOff
 // - ErrInvalidImageName
-func ValidatePodWaitingState(pod *Pod) error {
+// - ErrImagePull
+var InvalidPodWaitingStateErr = goerrors.New("invalid pod waiting state")
 
+func ValidatePodWaitingState(pod *Pod) error {
 	for _, invalidState := range InvalidPodWaitingStates {
 		status := &pod.Status
 		if checkWaitingReason(status, invalidState) {
-			return fmt.Errorf("one or more containers in pod are in %s", invalidState)
+			return fmt.Errorf("%s: one or more containers in pod are in %s", InvalidPodWaitingStateErr, invalidState)
```

Comment:
Maybe you wanted to use %w instead of %s here. Or maybe add .Error() to InvalidPodWaitingStateErr & replace fmt.Errorf by errors.New + fmt.Sprintf.

Same comment applies to other Errorf's added in this MR, specially in watcher.go, but I will concentrate it here to avoid repetitive comments.

Reply:
The current version of go that maestro is being built does not support this feature of using %w.